dapp/src/hooks/useTrackActivities.ts

```diff
@@ -0,0 +1,13 @@
+import { useEffect } from "react"
+import { setActivities } from "#/store/wallet"
+import { useActivitiesQuery } from "./tanstack-query"
+import { useAppDispatch } from "./store"
+
+export default function useTrackActivities() {
+  const { data } = useActivitiesQuery()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    dispatch(setActivities(data ?? []))
+  }, [data, dispatch])
```

Comment:
I think we can achieve the same using `queryClient.setQueryData` https://tanstack.com/query/latest/docs/reference/QueryClient#queryclientsetquerydata. Probably we will be able to remove activities from redux.